runtime/hydradx/src/xcm.rs

```diff
@@ -309,3 +309,92 @@ pub type LocalAssetTransactor = ReroutingMultiCurrencyAdapter<
 	OmnipoolProtocolAccount,
 	TreasuryAccount,
 >;
+
+/// A call filter for the XCM Transact instruction. This is a temporary measure until we properly
+/// account for proof size weights.
+///
+/// Calls that are allowed through this filter must:
+/// 1. Have a fixed weight;
+/// 2. Cannot lead to another call being made;
+/// 3. Have a defined proof size weight, e.g. no unbounded vecs in call parameters.
+pub struct SafeCallFilter;
+impl Contains<RuntimeCall> for SafeCallFilter {
+	fn contains(call: &RuntimeCall) -> bool {
+		#[cfg(feature = "runtime-benchmarks")]
+		{
+			if matches!(call, RuntimeCall::System(frame_system::Call::remark_with_event { .. })) {
+				return true;
+			}
+		}
+
+		// check the runtime call filter
+		if !CallFilter::contains(call) {
+			return false;
+		}
+
+		matches!(
+			call,
+			RuntimeCall::System(frame_system::Call::kill_prefix { .. } | frame_system::Call::set_heap_pages { .. })
```

Comment:
I think we might not want to allow this (It is probably done because of the switch to governance parachain should be able to do this but we don't have external entity that could call this AFAIK) cc @mrq1911

Reply:
My idea behind this filter is to list all possible extrinsics that pass requirements (fixed weight, no unbound vecs in params) and combine it with the runtime call filter. So the result are extrinsics that we want to allow, and can be executed